### PR TITLE
Feature/app header

### PR DESCRIPTION
**Pull request by m4manjesh**
Closes https://sce.myjetbrains.com/youtrack/issue/TD-1379/Create-new-react-ui-component-for-header

Figma
https://www.figma.com/proto/5qkWskMGpXY38REVIDcJnW/Landing-Page?page-id=0%3A1&node-id=3%3A7&starting-point-node-id=3%3A7&scaling=scale-down&show-proto-sidebar=1

- New app header including the sidebar, usermenu, mode toggle button and app launcher.
- Added IPG logo to svgicons
- Toggle Button style is handled separately (https://github.com/IPG-Automotive-UK/react-ui/pull/526)
- Current user menu UI is fine for now as discussed with @Sowbhagya-ipg

### CODE REVIEW

**Comment by m4manjesh** [comment]
> @m4manjesh can you link the latest figma prototype. I had a look for one, but it doesn't look like what you've implemented so I assume there is a newer version.

Added in the description.

**Comment by dmbarry86** [comment]
@m4manjesh have you tried integrating this with an app yet to check it looks and works as expected? If not, please do a pre-release build now and test it out so we can confirm all is ok. Send me a link to an app that conmtians it when done.

**Comment by dmbarry86** [comment]
Why is there a sidebar shown in the app header story? Is the sidebar actually part of the component? I don't think it should be.

image

**Comment by dmbarry86** [comment]
The APP NAME text is not aligned with VIRTO text. Also looks slightly different font or size.

Figma:
image


This branch:
image

**Comment by dmbarry86** [comment]
Prop names should be camelCase. Both logo link props are wrong. I also don't think you need a prop for IPG link. That should just be fixed to IPG website https://ipg-automotive.com/

image

**Comment by dmbarry86** [comment]
Which is mode a prop? Surely this is internal to the component and is set by the dark/light toggle switch that is part of the component. What else would be setting this in an app?

image

**Comment by dmbarry86** [comment]
Why do we need these props to show/hide these items? When would we ever hide them? Is there a particular app that needs them hidden?

image

**Comment by dmbarry86** [comment]
Is this the intended design for disabled buttons in dark mode to have the light grey transparent background? I don't see anything in Figma with disabled buttons so just checking. @Sowbhagya-ipg to confirm.

image

**Comment by dmbarry86** [comment]
The icons are smaller compared to Figma.

Figma (67px):
Screenshot 2023-02-28 at 17 01 44

This branch (54.32px):
image

**Comment by dmbarry86** [comment]
The app bar background color does not match Figma. Also note that the app text is black and doesn't match the color of the VIRTO text.

Figma:
image

This branch:
image

**Comment by dmbarry86** [comment]
Is this the expected design to show two menu buttons when the sidebar is collapsed? This doesn't feel nice to me. I don't see anything in Figma for this. @Sowbhagya-ipg to confirm expected behaviour here. Maybe we don't need to collapse sidebar anymore?

image

**Comment by dmbarry86** [comment]
Figma shows hover effect on each app button but this branch doesn't.

Figma:
https://user-images.githubusercontent.com/8976377/221927024-07f9a5ad-5a54-4ea7-8115-4420cf8f4352.mov

This branch:
https://user-images.githubusercontent.com/8976377/221927153-6828ed9a-ea68-4659-bbee-24f51cfa6bbc.mov

File: src/AppHeader/AppHeader.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const defaultInputs = {
+  onChangePassword: jest.fn(),
+  onLogout: jest.fn(),
+  onModeChange: jest.fn(),
+  username: "Joe Bloggs"
+};
+
+// test app name in the component
+describe("AppHeader", () => {
+  test("should render app name", () => {
```
Comment (dmbarry86) [inline src/AppHeader/AppHeader.test.js:16]: The tests for the app header component are somewhat lacking in coverage...

We can do better here. Please create a separate task to improve this after the code freeze.

**Review by dmbarry86 — CHANGES_REQUESTED** [review]
A few items to address. If you are planning to handle some of them as separate PRs then please leave comment against my respective comment so we keep track.

**Comment by dmbarry86** [comment]
> Why is there a sidebar shown in the app header story? Is the sidebar actually part of the component? I don't think it should be

We just had a call to discuss this. Agreed that it's useful to have a top level wrapper component that can render header, sidebar and content. However, that component is not AppHeader. We agreed to rename this component to AppLayout or such like and then have props for sidebarItems and content. At the moment, the current implementation doesn't allow any actual app content to be rendered. This AppLayout component can then contain all of the code we currently have in Layout of each app, including the ThemeProvider wrapping.

**Comment by Sowbhagya-ipg** [comment]
> Is this the expected design to show two menu buttons on the left when the sidebar is collapsed? This doesn't feel nice to me. I don't see anything in Figma for this. @Sowbhagya-ipg to confirm expected behaviour here. Maybe we don't need to collapse sidebar anymore?
> 
> image

We need the collapse Menu. Here, the position of icons has to be changed. More apps menu icon (9 dots) on the left most. @m4manjesh @dmbarry86

**Comment by Sowbhagya-ipg** [comment]
> Is this the intended design for disabled buttons in dark mode to have the light grey transparent background? I don't see anything in Figma with disabled buttons so just checking. @Sowbhagya-ipg to confirm.
> 
> image

This is fine.

**Comment by m4manjesh** [comment]
> @m4manjesh have you tried integrating this with an app yet to check it looks and works as expected? If not, please do a pre-release build now and test it out so we can confirm all is ok. Send me a link to an app branch that contains it when done and I will use that as part of the review.

I have done a pre-release and tested it in MM with some default data. This is the branch, please take a look https://github.com/IPG-Automotive-UK/model-manager/tree/feature/layout-update

**Comment by m4manjesh** [comment]
> The APP NAME text is not vertically aligned with VIRTO text. Also looks slightly different font or size.
> 
> Figma: image
> 
> This branch: image

Done and can test in MM branch.

**Comment by m4manjesh** [comment]
> User menu does not align with Figma. Neither does the dark/light switch.
> 
> Figma: image
> 
> This branch: image

@nikitharani is working on this. https://github.com/IPG-Automotive-UK/react-ui/pull/526

**Comment by m4manjesh** [comment]
> Done and can test in MM branch.

Done and can test in MM branch.

**Comment by m4manjesh** [comment]
> Why do we need these props to show/hide these items? When would we ever hide them? Is there a particular app that needs them hidden?
> 
> image

No, removed as discussed. Can test in MM branch.

**Comment by m4manjesh** [comment]
@dmbarry86  I am done with the changes. Had a chat with @Sowbhagya-ipg about the AppLauncher dark mode hover effect, she is fine with the new hover styles I have added.

**Review by dmbarry86 — CHANGES_REQUESTED** [review]
Almost there. Few small styling issues as mentioned in chat.

- [x] Spacing between top of sidebar item and the app bar.
- [x] APP NAME text height/position not aligned with VIRTO image.
- [ ] Small jumping in app bar height as I resize window. May just be Storybook issue in which case not too much of a problem but just double check in an app itself.

**Comment by m4manjesh** [comment]
> Almost there. Few small styling issues as mentioned in chat.
> 
> * [x]  Spacing between top of sidebar item and the app bar.
> * [x]  APP NAME text height/position not aligned with VIRTO image.
> * [x]  Small jumping in app bar height as I resize window. May just be Storybook issue in which case not too much of a problem but just double check in an app itself.

As mentioned in the chat the jumping is only in storybook. The updated layout in MM can find here, please take a look.
https://github.com/IPG-Automotive-UK/model-manager/tree/feature/layout-update

**Pull request merged**